src/main/java/net/minestom/server/network/packet/server/play/ResourcePackSendPacket.java

```diff
@@ -56,4 +63,23 @@ public void read(@NotNull BinaryReader reader) {
     public int getId() {
         return ServerPacketIdentifier.RESOURCE_PACK_SEND;
     }
+
+    @Override
+    public @NotNull Collection<Component> components() {
+        List<Component> components = new ArrayList<>();
+        if (forcedMessage != null) {
+            components.add(forcedMessage);
+        }
+        return components;
```

Comment:
Saves an allocation I suppose!
```suggestion
        if (forcedMessage != null) {
            return Collections.singleton(forcedMessage);
        } else {
            return Collections.emptyList();
        }
```